core/src/proto/types.rs

```diff
@@ -1002,7 +1033,7 @@ impl Tx {
         match &self.header.tx_type {
             // verify signature and extract signed data
             TxType::Wrapper(wrapper) => {
-                self.verify_signature(&wrapper.pk, &self.header_hash())
+                self.verify_signature(&wrapper.pk, &[self.header_hash()])
```

Comment:
Should we validate the signature over the header and also all of the sections? I see that in `sign_wrapper` we initialize the vector of hashes with the header hash and then push the hashes of all the sections included in the tx. I imagined we would the same here

Reply:
Good point, thanks. Currently we're just checking for a valid signature over the header, which indirectly implies a check that all sections targetted in the signature exist (i.e. haven't been removed or tampered with). But we should check that all sections in a transaction are signed over - this would prevent the addition of new sections to a transaction. Let me work on that.